lib/dockerregistry/inventory.go

```diff
@@ -255,6 +279,76 @@ func ParseManifestsFromDir(
 	return mfests, nil
 }
 
+// ValidateThinManifestDirectoryStructure enforces a particular directory
+// structure for thin manifests. Most importantly, it requires that if a file
+// named "foo/manifests/bar/promoter-manifest.yaml" exists, that a corresponding
+// file named "foo/images/bar/promoter-manifest.yaml" must also exist.
+//
+// nolint[gocyclo]
+func ValidateThinManifestDirectoryStructure(
+	dir string,
+) error {
+	// First, enforce that there are directories named "images" and "manifests".
+	dirExists, err := directoryExists(filepath.Join(dir, "images"))
+	if !dirExists || err != nil {
+		return err
```

Comment:
Blocker (?): If !dirExists but err==nil, we might return no error?  Though it is an error I believe.